### PR TITLE
[CALCITE-6265] Type coercion is failing for numeric values in prepared statements (follow-up)

### PR DESCRIPTION
**Pull request by rubenada**
Address some issues on the original patch (https://github.com/apache/calcite/commit/4e6a320bf476c4b3c313d86f8ff3c6edf4c2a578 , via PR https://github.com/apache/calcite/pull/3687 )
See details in CALCITE-6265

### CODE REVIEW

**Comment by rubenada** [comment]
@tindzk , could you please take a look at this PR, which addresses some issue from CALCITE-6265? (more info in the Jira ticket)

**Comment by tindzk** [comment]
The changes look good to me and your approach is a lot cleaner. Thanks for looking into it.

File: core/src/main/java/org/apache/calcite/adapter/enumerable/EnumUtils.java [hunk 1]
```diff
@@ -354,6 +354,9 @@ public static Expression convert(Expression operand, Type fromType,
     if (!Types.needTypeCast(fromType, toType)) {
       return operand;
     }
+
+    // TODO consider using Expressions#convertChecked to throw exception in case of overflow
```
Comment (asolimando) [inline core/src/main/java/org/apache/calcite/adapter/enumerable/EnumUtils.java:358]: @rubenada we might want to file a ticket for this and replace `TODO` with the ticket reference

**Review by asolimando — APPROVED** [review]
LGTM, left only a small comment, thanks @rubenada for looking into this

**Comment by rubenada** [comment]
Thanks for the feedback. I have created [CALCITE-6366](https://issues.apache.org/jira/browse/CALCITE-6366), and reference it on the PR code.
I'll merge in the next 24h (if there's no problem on CI), if no further remarks arrive

**Review by mihaibudiu — APPROVED** [review]

**Pull request merged**